### PR TITLE
OLS - additional regression statistics using linear algebra

### PR DESCRIPTION
**Pull request by rjenc29**
Morning, here's a smattering of OLS things...

### CODE REVIEW

**Comment by rjenc29** [comment]
ah yes, tests require statsmodels...

File: fastats/maths/ols.py [hunk 1]
```diff
@@ -32,6 +33,124 @@ def ols_qr(A, b):
...
 
 
+def add_intercept(A):
+    """
+    Adds an intercept (column of ones) to a supplied array A
+    such that the intercept is the first (leftmost) column
+    """
+    n = A.shape[0]
+    intercept = np.ones(n).reshape(n, 1)
+    return np.hstack([intercept, A])
```
Comment (dwillmer) [inline fastats/maths/ols.py]: Can we make this `hstack([intercept, A])` instead of calling from numpy?

File: fastats/maths/ols.py [hunk 2]
```diff
@@ -32,6 +33,124 @@ def ols_qr(A, b):
...
+    l = np.ones(n).reshape((n, 1))
+    return _hat(l)
+
+
+def sum_of_squared_residuals(A, b):
+    """
+    The sum of squared residuals
+    """
+    n = A.shape[0]
+    I = np.eye(n)
```
Comment (dwillmer) [inline fastats/maths/ols.py]: Same for `eye(n)` instead of `np.eye(n)`

File: tests/maths/test_ols_sklearn_diabetes.py [hunk 3]
```diff
@@ -53,6 +53,80 @@ def setUp(self):
...
+    assert np.allclose(output, expected)
+
+
+def test_t_statistic():
+    A, b, model = get_data_and_fit_statsmodel()
+    expected = model.tvalues
+    output = t_statistic(A, b)
+    assert np.allclose(output, expected)
+
+
```
Comment (dwillmer) [inline tests/maths/test_ols_sklearn_diabetes.py:129]: These tests are awesome, nice work!

**Comment by dwillmer** [comment]
This is awesome! I really like the r_squared :)

Just a couple of minor changes on the way the function calls are being made, then we'll merge

**Comment by dwillmer** [comment]
Also, as you've probably seen, i just added 'codacy' to the repo to get some code quality checks on each PR - yours passed first time :)

**Comment by rjenc29** [comment]
Changed the imports a little and re-committed

**Comment by dwillmer** [comment]
Just needs `pip install statsmodels` added to the travis.yml to get the tests to pass

**Comment by dwillmer** [comment]
Merged, that work is awesome, thanks dude!

**Comment by dwillmer** [comment]
@rjenc29 - After the excellent PRs, I've sent you an invite to join the fastats org. I've also added you to the `core` developers team, so you'll have write/merge permission on all the repos, and will get notified when anyone messages @fastats/core 

Congratulations 😃 🍾

**Pull request merged**